spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java

```diff
@@ -994,6 +1021,22 @@ void customizeJetty(ServerProperties serverProperties,
 			if (serverProperties.getMaxHttpPostSize() > 0) {
 				customizeMaxHttpPostSize(factory, serverProperties.getMaxHttpPostSize());
 			}
+
+			customizeConnectionTimeout(serverProperties, factory);
+		}
+		private void customizeConnectionTimeout(final ServerProperties serverProperties,
+													final JettyEmbeddedServletContainerFactory factory) {
+			factory.addServerCustomizers(new JettyServerCustomizer() {
+				@Override
+				public void customize(final Server server) {
+					for (org.eclipse.jetty.server.Connector connector : server.getConnectors()) {
+						if (connector instanceof ServerConnector) {
+							ServerConnector serverConnector = (ServerConnector) connector;
+							serverConnector.setSoLingerTime(serverProperties.getConnectionTimeout());
```

Comment:
The linger time for a socket is very different to the time it'll wait for a connection before being closed. When the socket is closed, the linger time causes the socket to block waiting for acknowledgement of the close from its peer.